hatchet/readers/hpctoolkit_reader_latest.py

```diff
@@ -0,0 +1,421 @@
+        self,
+        current_offset: int,
+        total_size: int,
+        parent: Node,
+        meta_db: bytes,
+        parent_time: int,
+    ) -> None:
+
+        final_offset = current_offset + total_size
+
+        while current_offset < final_offset:
+            (szChildren, pChildren, ctxId, _, lexicalType, nFlexWords) = safe_unpack(
+                "<QQLHBB", meta_db, current_offset
+            )
+
+            try:
+                my_time = self._summary_profile[ctxId][self._time_metric]
+            except:
+                my_time = None
+
+            if my_time is None:
+                current_offset += 32 + nFlexWords * 8  # TODO: can I move this up??
+                continue
+
+            if self._parent_percentage is not None and my_time / parent_time * 100.0 < self._parent_percentage:
+                current_offset += 32 + nFlexWords * 8  # TODO: can I move this up??
+                continue
+
+            if self._application_percentage is not None and my_time / self._total_execution_time * 100 < self._application_percentage:
+                current_offset += 32 + nFlexWords * 8  # TODO: can I move this up??
+                continue
+
+
+            frame = {
+                "id": ctxId,
+                "type": NODE_TYPE_MAPPING[lexicalType],
+                "depth": parent.frame["depth"] + 1,
+            }
+
+            if nFlexWords:
+                flex_offset = current_offset + 32
+
+                if lexicalType == 0:
+                    (pFunction,) = safe_unpack("<Q", meta_db, flex_offset)
+                    frame["name"] = self._parse_function(meta_db, pFunction)["name"]
+
+                elif lexicalType == 3:
+                    (pModule, offset) = safe_unpack("<QQ", meta_db, flex_offset)
+                    frame["name"] = (
+                        f"{self._parse_load_module(meta_db, pModule)['module_path']}:{offset}"
+                    )
+
+                else:
+                    (pFile, line) = safe_unpack("<QL", meta_db, flex_offset)
+                    frame["name"] = (
+                        f"{self._parse_source_file(meta_db, pFile)['file_path']}:{line}"
+                    )
+
+            node = Node(Frame(frame), parent=parent, hnid=ctxId)
+            parent.add_child(node)
+
+            node_value = {
+                "node": node,
+                "name": f"{frame['type']}: {frame['name']}",
+            }
+
+            if ctxId in self._summary_profile:
+                node_value.update(self._summary_profile[ctxId])
+
+            self._metrics_table.append(node_value)
+
+            if self._parse_depth is None or frame["depth"] < self._parse_depth:
+                
+                self._parse_context(
+                    pChildren,
+                    szChildren,
+                    node,
+                    meta_db,
+                    my_time,
+                )
+
+            current_offset += 32 + nFlexWords * 8  # TODO: can I move this up??
```

Comment:
Should this be here on its own (i.e., with no conditional)? Above, you guarded this "offset update" line with various `if` statements. As a result, if one of those conditions are satisfied, you will be doing a double increase of the offset. Just want to make sure that's intended.

Reply:
@ilumsden although there were multiple `if` statements above that were incrementing the `current_offset` variable, their increment was followed by a continue statement, which would prevent double increase of the offset. However, I updated this increment statement in my latest commit. I moved it up in the while loop, so I don't repeat its occurrence in multiple places. Check my latest commit.